### PR TITLE
copy sample use of warehouse to stage pattern with patch, updates fro…

### PR DESCRIPTION
**Pull request by chdietz**
…m recent changes , added more to  tests.

### CODE REVIEW

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Jdbc implementation of {@link SqlListExecutionRepository}.
+ */
+@Repository
+class JdbcSqlCopyWarehouseToStagingRepository implements SqlCopyWarehouseToStagingRepository {
+
+    private final  NamedParameterJdbcTemplate reportingJdbcTemplate;
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: nit ` `

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    private final NamedParameterJdbcTemplate warehouseJdbcTemplate;
+
+    @Autowired
+    public JdbcSqlCopyWarehouseToStagingRepository( NamedParameterJdbcTemplate reportingJdbcTemplate,
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: are the names letting Spring know which is which?

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        final MapSqlParameterSource[] stagingParams = new MapSqlParameterSource[result.size()];
+        int i = 0;
+        for (final Map<String, Object> row : result) {
+
+            final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
+            for (final String col : row.keySet()) {
+                parameterSource.addValue(col, row.get(col));
+            }
+            parameterSource.addValue("migrate_id", migrateId);
+            stagingParams[i++] = parameterSource;
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: I'd prefer to use a java List and then do the toArray thing if necessary when calling batchUpdate.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepositoryIT.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.context.annotation.Import;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.jdbc.Sql;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.jdbc.JdbcTestUtils.countRowsInTable;
+import static org.springframework.test.jdbc.JdbcTestUtils.countRowsInTableWhere;
+
+
```
Comment (malaffoon) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepositoryIT.java]: Please donate extra lines to undernourished developers.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/MigrateWarehouseToStagingSqlConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Configuration for the entities and corresponding SQLs (with an action and sql statement)
+ * user for migration from staging to reporting
+ */
+@Component
+@ConfigurationProperties(prefix = "sql.warehouseToStage")
+public class MigrateWarehouseToStagingSqlConfiguration {
```
Comment (malaffoon) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/MigrateWarehouseToStagingSqlConfiguration.java]: Can we reuse this? It is the same class with a different properties prefix, right?

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java (hunk 2 above)
Comment (agorina) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: Yes.

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java (hunk 1 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: Pas de format

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java (hunk 3 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/repository/impl/JdbcSqlCopyWarehouseToStagingRepository.java]: good idea

File: migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/MigrateWarehouseToStagingSqlConfiguration.java (hunk 5 above)
Comment (chdietz) [inline migrate-reporting/src/main/java/org/opentestsystem/rdw/ingest/migrate/reporting/MigrateWarehouseToStagingSqlConfiguration.java]: I'll look into this.

**Pull request merged**